proof_of_concept/definitions/identifier.py

```diff
@@ -0,0 +1,101 @@
+"""Identifier for various things in the DDM."""
+import re
+from typing import Any, cast, List, Type
+
+
+class Identifier(str):
+    """An identifier.
+
+    An Identifier is a string of any of the following forms:
+
+    1. party:<namespace>:<name>
+    2. party_collection:<namespace>:<name>
+    3. site:<namespace>:<name>
+    4. asset:<namespace>:<name>:<site_namespace>:<site_name>
+    5. asset_collection:<namespace>:<name>
+    6. result:<id_hash>
+
+    This class also accepts a single asterisk an an identifier, as it
+    is used as a wildcard in rules.
+
+    See the Terminology section of the documentation for details.
```

Comment:
Nice documentation! This is the minimum that you expect here, and in those cases you need more info you can read the Terminology documentation, awesome!